ts/Extensions/Annotations/Controllables/ControllableEllipse.ts

```diff
@@ -0,0 +1,292 @@
+/* *
+ *
+ * Author: Pawel Lysy
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import type Annotation from '../Annotations';
+import type SVGElement from '../../../Core/Renderer/SVG/SVGElement';
+import AxisType from '../../../Core/Axis/AxisType';
+import BBoxObject from '../../../Core/Renderer/BBoxObject';
+import ControllableMixin from '../Mixins/ControllableMixin.js';
+import ControllablePath from './ControllablePath.js';
+import MockPointOptions from '../MockPointOptions';
+
+import U from '../../../Core/Utilities.js';
+
+const {
+    merge,
+    defined
+} = U;
+
+/* eslint-disable no-invalid-this, valid-jsdoc */
+
+/**
+ * A controllable ellipse class.
+ *
+ * @requires modules/annotations
+ *
+ * @private
+ * @class
+ * @name Highcharts.AnnotationControllableEllipse
+ *
+ * @param {Highcharts.Annotation} annotation an annotation instance
+ * @param {Highcharts.AnnotationsShapeOptions} options a shape's options
+ * @param {number} index of the Ellipse
+ */
```

Comment:
This needs to be directly on top of the class pattern, otherwise it gets consumed by TypeScripts declarations filter, because it sesms to be about the following interface.